src/main/java/net/glowstone/io/entity/AreaEffectCloudStore.java

```diff
@@ -0,0 +1,111 @@
+package net.glowstone.io.entity;
+
+import java.util.UUID;
+import java.util.stream.Collectors;
+import net.glowstone.entity.GlowAreaEffectCloud;
+import net.glowstone.inventory.GlowMetaPotion;
+import net.glowstone.util.nbt.CompoundTag;
+import org.bukkit.Bukkit;
+import org.bukkit.Color;
+import org.bukkit.Location;
+import org.bukkit.Particle;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.potion.PotionData;
+import org.bukkit.projectiles.ProjectileSource;
+
+class AreaEffectCloudStore extends EntityStore<GlowAreaEffectCloud> {
+
+    private static final String DURATION = "Duration";
+    private static final String COLOR = "Color";
+    private static final String REAPPLICATION_DELAY = "ReapplicationDelay";
+    private static final String WAIT_TIME = "WaitTime";
+    private static final String OWNER_UUID_LEAST = "OwnerUUIDLeast";
+    private static final String OWNER_UUID_MOST = "OwnerUUIDMost";
+    private static final String DURATION_ON_USE = "DurationOnUse";
+    private static final String RADIUS = "Radius";
+    private static final String RADIUS_ON_USE = "RadiusOnUse";
+    private static final String RADIUS_PER_TICK = "RadiusPerTick";
+    private static final String PARTICLE = "Particle";
+    private static final String POTION = "Potion";
+    private static final String EFFECTS = "Effects";
+
+    public AreaEffectCloudStore() {
+        super(GlowAreaEffectCloud.class, EntityType.AREA_EFFECT_CLOUD);
+    }
+
+    @Override
+    public void load(GlowAreaEffectCloud entity, CompoundTag tag) {
+        super.load(entity, tag);
+        handleIntIfPresent(tag, COLOR, rgb -> entity.setColor(Color.fromRGB(rgb)));
+        handleIntIfPresent(tag, DURATION, entity::setDuration);
+        handleIntIfPresent(tag, REAPPLICATION_DELAY, entity::setReapplicationDelay);
+        handleIntIfPresent(tag, WAIT_TIME, entity::setWaitTime);
+        // TODO: OWNER_UUID_LEAST, OWNER_UUID_MOST
+        handleIntIfPresent(tag, DURATION_ON_USE, entity::setDurationOnUse);
+        handleFloatIfPresent(tag, RADIUS, entity::setRadius);
+        handleFloatIfPresent(tag, RADIUS_ON_USE, entity::setRadiusOnUse);
+        handleFloatIfPresent(tag, RADIUS_PER_TICK, entity::setRadiusPerTick);
+        if (tag.isString(PARTICLE)) {
+            final String particle = tag.getString(PARTICLE);
+
+            try {
+                entity.setParticle(Particle.valueOf(particle));
+            } catch (IllegalArgumentException e) {
+                Bukkit.getServer().getLogger().warning(String.format(
```

Comment:
Could you use `.log(Level.WARNING, ...)`

Reply:
Done in aaadd0e2356f861f5a5090c2d63ee61f501c23f2.